File: 2021/01a.cpp
#include "utils.h"
#include <iostream>

int main(int argc, char** argv)
{
	if (argc < 2)
	{
		std::cerr << "Specify filename as 1st param." << std::endl;
		return 1;
	}

	std::ifstream file(argv[1]);

	uint32_t last = 0;
	uint32_t result = 0;
	ReadAllLines(file, [&last, &result](std::string const& line){
		uint32_t sample = atoi(line.c_str());

		result += sample > last;
		last = sample;
	});

	std::cout << result << std::endl;

	return 0;
}

File: 2021/01a.cs
using System;
using System.Linq;

class Program
{
	public static int Main(string[] args)
	{
		if (args.Length < 1)
		{
			Console.WriteLine("Specify file as 1st argument.");
			return 1;
		}

		uint last = (uint)1 << 31;
		uint result = (uint)System.IO.File.ReadLines(args[0]).Count(line => {
			uint sample = uint.Parse(line);
			bool res = sample > last;
			last = sample;
			return res;
		});

		Console.WriteLine(result);

		return 0;
	}
}
